phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixConnection.java

```diff
@@ -426,6 +430,7 @@ public Void call() throws SQLException {
                 return null;
             }
         }, Tracing.withTracing(this, "committing mutations"));
+        statementExecutionCounter = 0;
     }
```

Comment:
Instead of mucking with the MutationState methods and having to add a setter for the state, how about this (part 1 of 3)?
{code}
        // In PhoenixConnection constructor, have construction of MutationState
        // go through method on PhoenixConnection to enable tests to specialize
        // the method
        this.mutationState = newMutationState(maxSize);
        this.services.addConnection(this);

        // setup tracing, if its enabled
        this.sampler = Tracing.getConfiguredSampler(this);
        this.customTracingAnnotations = getImmutableCustomTracingAnnotations();
    }
    
    protected MutationState newMutationState(int maxSize) {
        return new MutationState(maxSize, this); 
    }